pkg/controllers/provisioning/scheduling/scheduler.go

```diff
@@ -61,33 +61,31 @@ func NewScheduler(ctx context.Context, kubeClient client.Client, machines []*Mac
 	s := &Scheduler{
 		ctx:                ctx,
 		kubeClient:         kubeClient,
-		machineTemplates:   machines,
+		nodeClaimTemplates: nodeClaimTemplates,
 		topology:           topology,
 		cluster:            cluster,
 		instanceTypes:      instanceTypes,
-		daemonOverhead:     getDaemonOverhead(machines, daemonSetPods),
+		daemonOverhead:     getDaemonOverhead(nodeClaimTemplates, daemonSetPods),
 		recorder:           recorder,
 		opts:               opts,
 		preferences:        &Preferences{ToleratePreferNoSchedule: toleratePreferNoSchedule},
-		remainingResources: map[string]v1.ResourceList{},
+		remainingResources: map[nodepoolutil.Key]v1.ResourceList{},
 	}
-	for _, provisioner := range provisioners {
-		if provisioner.Spec.Limits != nil {
-			s.remainingResources[provisioner.Name] = provisioner.Spec.Limits.Resources
-		}
+	for _, nodePool := range nodePools {
+		s.remainingResources[nodepoolutil.Key{Name: nodePool.Name, IsProvisioner: nodePool.IsProvisioner}] = v1.ResourceList(nodePool.Spec.Limits)
 	}
-	s.calculateExistingMachines(stateNodes, daemonSetPods)
+	s.calculateExistingNodeClaims(stateNodes, daemonSetPods)
 	return s
 }
 
 type Scheduler struct {
 	ctx                context.Context
-	newMachines        []*Machine
+	newNodeClaims      []*NodeClaim
 	existingNodes      []*ExistingNode
-	machineTemplates   []*MachineTemplate
-	remainingResources map[string]v1.ResourceList // provisioner name -> remaining resources for that provisioner
-	instanceTypes      map[string][]*cloudprovider.InstanceType
-	daemonOverhead     map[*MachineTemplate]v1.ResourceList
+	nodeClaimTemplates []*NodeClaimTemplate
+	remainingResources map[nodepoolutil.Key]v1.ResourceList               // (nodePool name, isProvisioner) -> remaining resources for that provisioner
```

Comment:
Probably can change both comments if you're gonna change one.
```suggestion
	remainingResources map[nodepoolutil.Key]v1.ResourceList               // (nodePool name, isProvisioner) -> remaining resources for that nodepool
```